xs2a-adapter-aspsp-registry/src/main/java/de/adorsys/xs2a/adapter/registry/LuceneAspspRepository.java

```diff
@@ -282,15 +285,50 @@ private Query getNameRegexpQuery(String name) {
     public List<Aspsp> findLike(Aspsp aspsp, String after, int size) {
         BooleanQuery.Builder queryBuilder = new BooleanQuery.Builder();
         if (aspsp.getName() != null) {
-            queryBuilder.add(getNameFuzzyQuery(aspsp.getName()), BooleanClause.Occur.SHOULD);
-            queryBuilder.add(getNameRegexpQuery(aspsp.getName()), BooleanClause.Occur.SHOULD);
+            queryBuilder.add(buildQueueWithPriority(getNameFuzzyQuery(aspsp.getName()), LOW_PRIORITY), BooleanClause.Occur.SHOULD);
+            queryBuilder.add(buildQueueWithPriority(getNameRegexpQuery(aspsp.getName()), LOW_PRIORITY), BooleanClause.Occur.SHOULD);
         }
-        if (aspsp.getBic() != null) {
-            queryBuilder.add(getBicQuery(aspsp.getBic()), BooleanClause.Occur.SHOULD);
-        }
-        if (aspsp.getBankCode() != null) {
-            queryBuilder.add(getBankCodeQuery(aspsp.getBankCode()), BooleanClause.Occur.SHOULD);
+        if (aspsp.getBic() != null && aspsp.getBankCode() != null) {
+            String bic = aspsp.getBic();
+            String bankCode = aspsp.getBankCode();
+
+            queryBuilder.add(buildQueueWithPriority(getBicAndBankCodeQuery(bic, bankCode), HIGH_PRIORITY), BooleanClause.Occur.SHOULD);
+            queryBuilder.add(buildQueueWithPriority(getBicAndEmptyBankCodeQuery(bic), MEDIUM_PRIORITY), BooleanClause.Occur.SHOULD);
+            queryBuilder.add(buildQueueWithPriority(getEmptyBicAndBankCodeQuery(bankCode), LOW_PRIORITY), BooleanClause.Occur.SHOULD);
+        } else {
+            if (aspsp.getBic() != null) {
+                queryBuilder.add(buildQueueWithPriority(getBicQuery(aspsp.getBic()), MEDIUM_PRIORITY), BooleanClause.Occur.SHOULD);
```

Comment:
Boost to 1 is a noop, isn't it?

Reply:
Yes, but I'm not 100% sure about the Lucene behavior if some queries contain no priority when others contain some.
Also, it seems to me that such mentioning increases code readability, as it is clear, that some of the queries are low priority ones, others - high, and these are - medium.

Reply:
You're saying you don't trust your own tests?
I don't think that such classification is a good idea. It makes adding new fields more difficult because the person has to understand that LOW is actually the baseline. In my opinion you should use boost only to increase the rank.

Reply:
`You're saying you don't trust your own tests?`
I think it will be wise from my side to leave this question unanswered.

`In my opinion you should use boost only to increase the rank.`
Ok, thanks, I got your message. I have a different opinion, that we should use boots for all the subqueries of the parent query if the current search requires the prioritization of the results. 

@spetrychenko , we need the third opinion here.

Reply:
Removed setting medium priorities for the queries within the `else` statement.